### ISSUE TITLE
NETCOREAPP2_0_OR_GREATER not defined when net5-windows + UseWindowsForms

### ISSUE THREAD

**Issue by CreateAndInject** [dotnet/roslyn#66123]
Compiler doesn't compile `Test1` when targeting net5-windows (>=.NETCore3)
```c#
#if NETCOREAPP2_0_OR_GREATER

class Test1
{

}
#endif

#if NETCOREAPP3_0_OR_GREATER
class Test2
{

}
#endif


namespace ConsoleApp_NETCore
{
    class Program
    {
        static void Main(string[] args)
        {
            Log("test");
        }

        static void Log(string msg)
        {
#if NETCOREAPP3_0_OR_GREATER
            System.Windows.Forms.MessageBox.Show(msg);
#else
            System.Console.WriteLine(msg);
#endif
        }
    }
}
```

```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
	  <UseWindowsForms>true</UseWindowsForms>
	  <!--<UseWindowsForms Condition="'$(TargetFramework)' != 'netcoreapp2.1'">true</UseWindowsForms>-->
	  <TargetFrameworks>netcoreapp2.1;netcoreapp3.0-windows;net5-windows;net6-windows</TargetFrameworks>
	  <LangVersion>preview</LangVersion>
  </PropertyGroup>

</Project>
```

I know .NETCore2.1 doesn't support WinForm, so I already add `#if` to prevent .NETCore2.1 compile WinForms codes

**Comment by CyrusNajmabadi** [comment]
What error do you get?

**Comment by CreateAndInject** [comment]
`class Test1` is not a part of net5-windows output

**Comment by CreateAndInject** [comment]
@CyrusNajmabadi Can you reproduce this bug?

**Comment by Youssef1313** [comment]
I don't think this is a Roslyn issue. I looked at how dotnet/sdk generates symbols for implicit `DefineConstants`, and the root cause for this issue is that .NET Core 2.0 isn't in `SupportedNETCoreAppTargetFramework` when building against `netcoreapp3.0-windows`:

![image](https://user-images.githubusercontent.com/31348972/209978316-63afaf3d-c721-4560-8afc-21f950e057e5.png)

I can't track down this further, but I think investigation should start at dotnet/sdk, and they can move the issue appropriately after investigation.

**Comment by Youssef1313** [comment]
I could track this down. The bug is actually on dotnet/wpf side:

https://github.com/dotnet/wpf/blob/bed334c8b18400e79ef7b0287965e9f642932155/packaging/Microsoft.NET.Sdk.WindowsDesktop/targets/Microsoft.NET.Sdk.WindowsDesktop.props#L130-L167

They remove few TFMs from `SupportedNETCoreAppTargetFramework` (including .NET Core 2.0), which then affects `<DefineConstants>`

@CreateAndInject As a workaround, could you see if setting `<_RemoveUnsupportedTargetFrameworksForWindowsDesktop>false</_RemoveUnsupportedTargetFrameworksForWindowsDesktop>` works?
